deployment/common/changeset/internal/solana/mcms.go

```diff
@@ -0,0 +1,101 @@
+	e deployment.Environment,
+	chain deployment.SolChain,
+	addressBook deployment.AddressBook,
+	config commontypes.MCMSWithTimelockConfigV2,
+) (*state.MCMSWithTimelockStateSolana, error) {
+	addresses, err := e.ExistingAddresses.AddressesForChain(chain.Selector)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get addresses for chain %v from environment: %w", chain.Selector, err)
+	}
+
+	chainState, err := state.MaybeLoadMCMSWithTimelockChainStateSolana(chain, addresses)
+	if err != nil {
+		return nil, fmt.Errorf("failed to load mcms with timelock solana chain state: %w", err)
+	}
+
+	// access controller
+	err = deployAccessControllerProgram(e, chainState, chain, addressBook)
+	if err != nil {
+		return nil, fmt.Errorf("failed to deploy access controller program: %w", err)
+	}
+	err = initAccessController(e, chainState, commontypes.ProposerAccessControllerAccount, chain, addressBook)
+	if err != nil {
+		return nil, fmt.Errorf("failed to deploy proposer access controller: %w", err)
+	}
+	err = initAccessController(e, chainState, commontypes.ExecutorAccessControllerAccount, chain, addressBook)
+	if err != nil {
+		return nil, fmt.Errorf("failed to deploy access controller: %w", err)
+	}
+	err = initAccessController(e, chainState, commontypes.CancellerAccessControllerAccount, chain, addressBook)
+	if err != nil {
+		return nil, fmt.Errorf("failed to deploy access controller: %w", err)
+	}
+	err = initAccessController(e, chainState, commontypes.BypasserAccessControllerAccount, chain, addressBook)
+	if err != nil {
+		return nil, fmt.Errorf("failed to deploy access controller: %w", err)
+	}
+
+	// mcm
+	err = deployMCMProgram(e, chainState, chain, addressBook)
+	if err != nil {
+		return nil, fmt.Errorf("failed to deploy mcm program: %w", err)
+	}
+	err = initMCM(e, chainState, commontypes.BypasserManyChainMultisig, chain, addressBook, &config.Bypasser)
+	if err != nil {
+		return nil, fmt.Errorf("failed to deploy bypasser mcm: %w", err)
+	}
+	err = initMCM(e, chainState, commontypes.CancellerManyChainMultisig, chain, addressBook, &config.Canceller)
+	if err != nil {
+		return nil, fmt.Errorf("failed to deploy canceller mcm: %w", err)
+	}
+	err = initMCM(e, chainState, commontypes.ProposerManyChainMultisig, chain, addressBook, &config.Proposer)
+	if err != nil {
+		return nil, fmt.Errorf("failed to deploy proposer mcm: %w", err)
+	}
+
+	// timelock
+	err = deployTimelockProgram(e, chainState, chain, addressBook)
+	if err != nil {
+		return nil, fmt.Errorf("failed to deploy timelock program: %w", err)
+	}
+	err = initTimelock(e, chainState, chain, addressBook, config.TimelockMinDelay)
+	if err != nil {
+		return nil, fmt.Errorf("failed to deploy timelock: %w", err)
+	}
+
+	err = setupRoles(chainState, chain)
+	if err != nil {
+		return nil, fmt.Errorf("failed to setup roles and ownership: %w", err)
+	}
+
+	err = transferOwnership(chainState, chain)
+	if err != nil {
+		return nil, fmt.Errorf("failed to setup roles and ownership: %w", err)
```

Comment:
```suggestion
		return nil, fmt.Errorf("???: %w", err)
```

or maybe just return err as we have good err message in `transferOwnership`

Reply:
resolved in 0a3666439ed159a93552ae9b9c550a3b90904fe0